easybuild/framework/easyconfig/tools.py

```diff
@@ -140,6 +141,225 @@ def find_resolved_modules(unprocessed, avail_modules, retain_all_deps=False):
 
     return ordered_ecs, new_unprocessed, new_avail_modules
 
+def toolchain_hierarchy_cache(func):
+    mydict = dict()
```

Comment:
missing docstring, and use `cache = {}` here

where did you get inspiration for this BTW?

I've never seen a cache implemented via a decorator function before... :-)